lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -231,6 +231,7 @@ def main():
                     params['ssh_key_data'] = f.read()
 
             if state == 'present':
+                params.pop('state')
```

Comment:
would it be safer to do `.pop('state', None)`? I feel skeptical that the source of the parameter will consistently be there.